lib/bridge/IrcHandler.js

```diff
@@ -652,10 +654,12 @@ IrcHandler.prototype.onKick = Promise.coroutine(function*(req, server,
             return;
         }
         let promises = matrixRooms.map((room) => {
-            req.log.info("Leaving (due to kick) room %s", room.getId());
-            return this.ircBridge.getAppServiceBridge().getIntent(
-                matrixUser.getId()
-            ).leave(room.getId());
+            return this.delayWhileInflight(room, matrixUser).then(() => {
+                req.log.info("Leaving (due to kick) room %s", room.getId());
+                return this.ircBridge.getAppServiceBridge().getIntent(
+                    matrixUser.getId()
+                ).leave(room.getId());
```

Comment:
The column limit is really driving me crazy. Not something for this PR, but we should consider raising it or removing it.